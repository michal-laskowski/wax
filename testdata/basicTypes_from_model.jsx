export function basicTypes_from_model(p) {
    return (
        <>
            <sample-01>{p.SomeStringArr}</sample-01>
            <sample-02>{p.SomeStringPtrArr}</sample-02>
            <sample-03>{p.PtrArr}</sample-03>
            <sample-04>{p.PtrArrPtr}</sample-04>
            <sample-05>{p.AliasToString}</sample-05>
            <sample-05a>
                {typeof p.AliasToString} - type alias will be object in goja
            </sample-05a>
            <sample-05b>
                {typeof p.AliasToString.SomeAliasTrueMethod}
            </sample-05b>
            <sample-05c>
                {typeof p.AliasToString.SomeAliasFalseMethod}
            </sample-05c>
            <sample-06>{p.Ballance}</sample-06>
            <sample-07>{p.Deposit}</sample-07>
            <sample-08>{p.OtherDummySimple}</sample-08>
            <sample-09>{p.OtherDummyMaps}</sample-09>
            <sample-10>{p.OtherDummySimpleGeneric}</sample-10>
            <sample-11>{p.OtherDummyBasicTypes}</sample-11>
            <sample-12>{p.DummySimple}</sample-12>
            <sample-13>{p.GoPublicThatDoesNotExist}</sample-13>
            <sample-13>{p.goPrivateField}</sample-13>
        </>
    );
}
