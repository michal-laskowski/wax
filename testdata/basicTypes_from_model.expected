
    <sample-01>[str1 str2]</sample-01>
    <sample-02>[]</sample-02>
    <sample-03>[str1 str2]</sample-03>
    <sample-04>[]</sample-04>
    <sample-05>some-str-value-in-alias</sample-05>
    <sample-05a>object - type alias will be object in goja</sample-05a>
    <sample-05b>function</sample-05b>
    <sample-05c>function</sample-05c>
    <sample-06>339999995214436420000000000000000000000</sample-06>
    <sample-07>170000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000</sample-07>
    <sample-08>{OtherDummySimple-str-value}</sample-08>
    <sample-09>{map[] map[] map[] map[] map[] map[]}</sample-09>
    <sample-10>{888888}</sample-10>
    <sample-11>{false OtherDummyBasicTypes-str 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 (0+0i) (0+0i)}</sample-11>
    <sample-12>{DummySimple-Embedded-str-value}</sample-12>
    <sample-13></sample-13>
    <sample-13></sample-13>